lib/elastic_apm/context.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'elastic_apm/context/request/url'
+require 'elastic_apm/context/request/socket'
+require 'elastic_apm/context/request/apply_rack_env'
+
+module ElasticAPM
+  # @api private
+  class Context
+    # @api private
+    class Request
+      def initialize
+        @socket = {}
+        @headers = {}
+        @cookies = {}
+        @env = {}
```

Comment:
I think you are missing `url` here.

Reply:
I don't think there's any reason to initialize these anymore.